torch/_dynamo/variables/builder.py

```diff
@@ -1686,24 +1686,36 @@ def wrap_module(self, value: torch.nn.Module):
             self.install_guards(GuardBuilder.TYPE_MATCH)
             if torch._dynamo.config.inline_inbuilt_nn_modules:
                 freezing = is_parameter_freezing()
-                # Guard against the case where user may overwrite named parameters
-                # / named buffers
-                # NOTE: This is not likely to happen but worth guarding to avoid
-                # exception
-                if (
-                    callable(value.named_parameters)
-                    and value.named_parameters.__func__
-                    is og_module_named_parameters_fn_ptr
-                ):
-                    for _, p in value.named_parameters():
-                        self.mark_static_input(p, guard=freezing)
-
-                if (
-                    callable(value.named_buffers)
-                    and value.named_buffers.__func__ is og_module_named_buffers_fn_ptr
-                ):
-                    for _, b in value.named_buffers():
-                        self.mark_static_input(b, guard=freezing)
+
+                try:  # catch TypeErrors from unserializable nn modules
+                    # Guard against the case where user may overwrite named parameters
+                    # / named buffers
+                    # NOTE: This is not likely to happen but worth guarding to avoid
+                    # exception
+                    if (
+                        callable(value.named_parameters)
+                        and value.named_parameters.__func__
+                        is og_module_named_parameters_fn_ptr
+                    ):
+                        for _, p in value.named_parameters():
```

Comment:
You should put the exception catching narrowly around where we're calling into user code (otherwise, if we accidentally cause a type error somewhere else in the code, it'll be incorrectly marked as a user input.

Reply:
Fixed.